### ISSUE TITLE
Description of `sending` in Flow-Sensitive Isolation Analysis needs more elaboration

### ISSUE THREAD

**Issue by bjo0** [apple/swift-migration-guide#123]
The text on it is as follows:

> Region-based isolation works without any code changes. But a function’s parameters and return values can also explicitly state that they support crossing domains using this mechanism.
```swift
func populate(island: Island, with chicken: sending Chicken) async {
    await island.adopt(chicken)
}
```
> The compiler can now provide the guarantee that at all call sites, the chicken parameter will never be subject to unsafe access. This is a relaxing of an otherwise significant constraint. Without sending, this function would only be possible to implement by requiring that Chicken first conform to Sendable.

Based on this, I don't feel like I understand how to reason about using `sending`. Is the compiler still going to guarantee safety, or is this a foot gun? If it still guarantees safety, how is it any different than omitting `sending`?

**Comment by mattmassicotte** [comment]
Yes, I agree. There is no unsafety here. But, this example in isolation doesn't make it clear enough that **without** `sending`, this code will not compile because the compile cannot guarantee that `chicken` can be transferred across domains.
